src/Network/OAuth/OAuth2/Internal.hs

```diff
@@ -63,12 +67,62 @@ instance FromJSON OAuth2Token where
 instance ToJSON OAuth2Token where
     toEncoding = genericToEncoding defaultOptions { fieldLabelModifier = camelTo2 '_' }
 
+-- | Possible OAuth error types
+-- | Not all of them can appear everywhere, but this is good enough for now.
```

Comment:
Two groups should work nicely. Not sure which errors can appear where though, depending on which services follow the spec how closely.